dds/DCPS/transport/tcp/TcpDataLink.cpp

```diff
@@ -331,14 +331,107 @@ OpenDDS::DCPS::TcpDataLink::send_graceful_disconnect_message()
   this->send_i(send_element, false);
 }
 
-void OpenDDS::DCPS::TcpDataLink::set_release_pending(bool flag)
+void
+OpenDDS::DCPS::TcpDataLink::set_release_pending(bool flag)
 {
   this->release_is_pending_ = flag;
 }
 
-bool OpenDDS::DCPS::TcpDataLink::is_release_pending() const
+bool
+OpenDDS::DCPS::TcpDataLink::is_release_pending() const
 {
   return this->release_is_pending_.value();
 }
 
+bool
+OpenDDS::DCPS::TcpDataLink::handle_send_request_ack(TransportQueueElement* element)
+{
+  if (DCPS_debug_level >= 1) {
+    ACE_DEBUG((LM_DEBUG, "(%P|%t) TcpDataLink::handle_send_request_ack() sequence number %q\n",
+      element->sequence().getValue()));
+  }
+
+  ACE_Guard<ACE_SYNCH_MUTEX> guard(pending_request_acks_lock_);
+  pending_request_acks_.push_back(element);
```

Comment:
Are there cases where this vector needs to be cleaned up if the DataLink is shut down before it's acknowledged?